services/minespace-web/src/components/dashboard/mine/reports/Reports.tsx

```diff
@@ -0,0 +1,191 @@
+  fetchMineReports: ActionCreator<typeof fetchMineReports>;
+  openModal: (any?) => void;
+  closeModal: (any?) => void;
+}
+
+export const Reports: FC<ReportsProps> = ({
+  mine,
+  mineReports,
+  mineReportDefinitionOptions,
+  ...props
+}) => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [report, setReport] = useState(null);
+
+  useEffect(() => {
+    props.fetchMineReports(mine.mine_guid).then(() => {
+      setIsLoaded(true);
+    });
+  }, []);
+
+  useEffect(() => {
+    const reportsSubmitted = mineReports.filter(
+      (report) => report.mine_report_submissions.length > 0
+    ).length;
+    const reportsDue = mineReports.filter(
+      (report) => report.mine_report_submissions.length === 0 && report.due_date
+    ).length;
+  }, [mineReports]);
+
+  const handleAddReport = (values) => {
+    const formValues = values;
+    if (values.mine_report_submissions && values.mine_report_submissions.length > 0) {
+      formValues.received_date = moment().format("YYYY-MM-DD");
+    }
+    return props
+      .createMineReport(mine.mine_guid, formValues)
+      .then(() => props.closeModal())
+      .then(() => props.fetchMineReports(mine.mine_guid));
+  };
+
+  const handleEditReport = (values) => {
+    if (!values.mine_report_submissions || values.mine_report_submissions.length === 0) {
+      props.closeModal();
+      return;
+    }
+
+    let payload: any = {
+      mine_report_submissions: [
+        ...values.mine_report_submissions,
+        {
+          documents:
+            values.mine_report_submissions[values.mine_report_submissions.length - 1].documents,
+        },
+      ],
+    };
+
+    if (
+      !report.received_date &&
+      values.mine_report_submissions &&
+      values.mine_report_submissions.length > 0
+    ) {
+      payload = { ...payload, received_date: moment().format("YYYY-MM-DD") };
+    }
+
+    return props
+      .updateMineReport(mine.mine_guid, report.mine_report_guid, payload)
+      .then(() => props.closeModal())
+      .then(() => props.fetchMineReports(mine.mine_guid));
+  };
+
+  const openAddReportModal = (event) => {
+    event.preventDefault();
+    props.openModal({
+      props: {
+        onSubmit: handleAddReport,
+        title: "Add Report",
+        mineGuid: mine.mine_guid,
+        width: "40vw",
+      },
+      content: modalConfig.ADD_REPORT,
+    });
+  };
+
+  const openEditReportModal = (event, report) => {
+    event.preventDefault();
+    setReport(report);
+    props.openModal({
+      props: {
+        onSubmit: handleEditReport,
+        title: `Add Documents to: ${report.report_name}`,
+        width: "40vw",
+        mineGuid: mine.mine_guid,
+        mineReport: report,
+      },
+      content: modalConfig.EDIT_REPORT,
+    });
+  };
+
+  return (
+    <Row>
+      <Col span={24}>
+        <Row>
+          <Col span={24}>
+            <div className="submit-report-button">
+              <AuthorizationWrapper>
```

Comment:
What's the authorization wrapper looking for? I thought it wanted a permission.

Reply:
🤔 Didn't mess with the guts of this at all, so off the top of my head I'm not sure.  I'll take a peek.

Reply:
Hmmm, looks like this wrapper is meant more to prevent use of certain functions if a user is signed in with an IDIR.  So it just makes sure they're using a BCEID.